store/tikv/gcworker/gc_worker_test.go

```diff
@@ -875,6 +882,69 @@ func (s *testGCWorkerSuite) TestResolveLockRangeMeetRegionCacheMiss(c *C) {
 	c.Assert(scanCnt, Equals, 1)
 }
 
+func (s *testGCWorkerSuite) TestResolveLockRangeMeetRegionEnlargeCausedByRegionMerge(c *C) {
+	var (
+		firstAccess    = true
+		firstAccessRef = &firstAccess
+		resolvedLock   [][]byte
+	)
+
+	// key range: ['' - 'm' - 'z']
+	region2 := s.cluster.AllocID()
+	newPeers := []uint64{s.cluster.AllocID(), s.cluster.AllocID(), s.cluster.AllocID()}
+	s.cluster.Split(s.initRegion.regionID, region2, []byte("m"), newPeers, newPeers[0])
+
+	// init a, b lock in region1 and o, p locks in region2
+	s.gcWorker.testingKnobs.scanLocks = func(key []byte, regionID uint64) []*tikv.Lock {
+		if regionID == s.initRegion.regionID {
+			return []*tikv.Lock{{Key: []byte("a")}, {Key: []byte("b")}}
+		}
+		if regionID == region2 {
+			return []*tikv.Lock{{Key: []byte("o")}, {Key: []byte("p")}}
+		}
+		return []*tikv.Lock{}
+	}
+
+	s.gcWorker.testingKnobs.resolveLocks = func(locks []*tikv.Lock, regionID tikv.RegionVerID) (ok bool, err error) {
+		if regionID.GetID() == s.initRegion.regionID && *firstAccessRef {
+			*firstAccessRef = false
+			// merge region2 into region1 and return EpochNotMatch error.
+			mCluster := s.cluster.(*mocktikv.Cluster)
+			mCluster.Merge(s.initRegion.regionID, region2)
+			regionMeta, _ := mCluster.GetRegion(s.initRegion.regionID)
+			s.store.GetRegionCache().OnRegionEpochNotMatch(
+				tikv.NewNoopBackoff(context.Background()),
+				&tikv.RPCContext{Region: regionID, Store: &tikv.Store{}},
+				[]*metapb.Region{regionMeta})
+			// also let region1 contains all 4 locks
+			s.gcWorker.testingKnobs.scanLocks = func(key []byte, regionID uint64) []*tikv.Lock {
+				if regionID == s.initRegion.regionID {
+					return []*tikv.Lock{
+						{Key: []byte("a")},
+						{Key: []byte("b")},
+						{Key: []byte("o")},
+						{Key: []byte("p")},
```

Comment:
You can compare the `key` parameter with these 4 locks, so that there should be 4 locks at line 942.